### DISCUSSION TITLE
How to use CustomCast with different parameters in same model

### DISCUSSION THREAD

**Discussion by ApetersDev** [CodeDredd/pinia-orm#1893] [Q&A]
I have created a custom cast which needs a parameter like it´s described here https://pinia-orm.codedredd.de/guide/model/casts#with-parameters

It works great for different attributes of a model if the cast is used with the same parameter but it is not possible to use the same cast with different parameters. The last cast definition wins and every cast will use the same parameter. How can we use the same cast with different parameters in the same model?

```
//will work because parameters are the same
static casts() {
  return {
    name: CustomCast.withParameters({ type: 'string' }),
    age: CustomCast.withParameters({ type: 'string' }),
  }
}
 
//will not work, because parameters for name will be overwritten with { type: 'number' }
static casts() {
  return {
    name: CustomCast.withParameters({ type: 'string' }),
    age: CustomCast.withParameters({ type: 'number' }),
  }
}
```

**Comment by CodeDredd** [comment]
Hmmm this seems like a bug. Going to test it in a unit test.
